bin/rucio

```diff
@@ -1118,6 +1137,19 @@ def download(args):
     return SUCCESS
 
 
+def _get_rse_for_pfn(replicas, pfn) -> "Optional[str]":
+    # Check each rse in the replica list for the pfn. If no pfn is found, returns None.
+    # If it is found, stop the generator and return the item.
+    download_rse = next((
+        next((
+            rse for rse in replica['rses']
+            if pfn in replica['rses'][rse]
+        ), None)
+        for replica in replicas
+    ), None)
```

Comment:
This works fine, but I think that the outer `next()` is not necessary over a regular `for` loop (since it does not have a condition tied to it, it's just iterating).

For readability, I think this would be easier to understand:
```python
def _get_rse_for_pfn(replicas, pfn):
    # Check each rse in the replica list for the pfn. If no pfn is found, returns None.
    # If it is found, stop the generator and return the item.
    for replica in replicas:
        try:
            download_rse = next((
                rse for rse in replica['rses']
                if pfn in replica['rses'][rse]
            ))
        except StopIteration:
            continue
        else:
            return download_rse
```

This is just my opinion though, I think it's also perfectly fine to keep the current version

Reply:
Yes, please still change this @voetberg